src/tasks/withdraw.ts

```diff
@@ -0,0 +1,396 @@
+      withdrawal.balance,
+      withdrawal.token.decimals ?? DAI_DECIMALS,
+      18,
+    ),
+    symbol: withdrawal.token.symbol ?? "unknown token",
+  };
+}
+
+function displayWithdrawals(withdrawals: Withdrawal[]) {
+  const formattedWithdtrawals = withdrawals.map(formatWithdrawal);
+  const order = ["address", "value", "balance", "symbol"] as const;
+  const header = {
+    address: "address",
+    value: "value (usd)",
+    balance: "balance",
+    symbol: "symbol",
+  };
+  console.log(chalk.bold("Amounts to withdraw:"));
+  displayTable(header, formattedWithdtrawals, order, {
+    value: { align: Align.Right },
+    balance: { align: Align.Right, maxWidth: 30 },
+    symbol: { maxWidth: 20 },
+  });
+  console.log();
+}
+
+async function formatGasCost(
+  amount: BigNumber,
+  network: string,
+): Promise<string> {
+  switch (network) {
+    case "mainnet": {
+      const value = await oneinchUsdValue(
+        { symbol: "ETH", address: ONEINCH_ETH_FLAG },
+        amount,
+        "mainnet",
+      );
+      return `${utils.formatEther(amount)} ETH (${formatUsdValue(value)} USD)`;
+    }
+    case "xdai":
+      return `${utils.formatEther(amount)} XDAI`;
+    default:
+      return `${utils.formatEther(amount)} ETH`;
+  }
+}
+
+async function prompt(message: string) {
+  const rl = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout,
+  });
+  const response = await new Promise<string>((resolve) =>
+    rl.question(`${message} (y/N) `, (response) => resolve(response)),
+  );
+  return "y" === response.toLowerCase();
+}
+
+const setupWithdrawTask: () => void = () =>
+  task("withdraw", "Withdraw funds from the settlement contract")
+    .addOptionalParam(
+      "minValue",
+      "If specified, sets a minimum USD value required to withdraw the balance of a token.",
+      "0",
+      types.string,
+    )
+    .addParam("receiver", "The address receiving the withdrawn tokens.")
+    .addFlag(
+      "dryRun",
+      "Just simulate the settlement instead of executing the transaction on the blockchain.",
+    )
+    .setAction(
+      async (
+        { minValue, dryRun, receiver: inputReceiver },
+        hre: HardhatRuntimeEnvironment,
+      ) => {
+        const receiver = utils.getAddress(inputReceiver);
+        const [authenticator, settlement, [solver]] = await Promise.all([
+          getDeployedContract("GPv2AllowListAuthentication", hre),
+          getDeployedContract("GPv2Settlement", hre),
+          hre.ethers.getSigners(),
+        ]);
+
+        if (!(await authenticator.isSolver(solver.address))) {
+          const message =
+            "Current account is not a solver. Only a solver can withdraw funds from the settlement contract.";
+          if (!dryRun) {
+            throw Error(message);
+          } else {
+            console.warn(message);
+          }
+        }
+
+        const tokens = await getAllTradedTokens(settlement);
+
+        // TODO: add eth withdrawal
```

Comment:
Not from standard orders. It could come only from positive slippage in an interaction receiving ETH or from someone sending eth to the contract directly.
So far none of our contracts has any ETH, so I thought it was fine to ignore it, but I still thought it would be good to mention it somewhere.

Reply:
Also, in principle there is nothing preventing a solver from just `unwrap`-ing some WETH. I also don't think we need to support it until the issue arises.